app/models/annotation/AnnotationService.scala

```diff
@@ -42,8 +42,9 @@ object AnnotationService
   private def selectSuitableTeam(user: UserSQL, dataSet: DataSet)(implicit ctx: DBAccessContext): Fox[ObjectId] = {
     (for {
       userTeamIds <- user.teamIds
+      userTeamIdsBson <- Fox.serialCombined(userTeamIds)(_.toBSONObjectId.toFox)
     } yield {
-      val selectedTeamOpt = dataSet.allowedTeams.intersect(userTeamIds).headOption
+      val selectedTeamOpt = dataSet.allowedTeams.intersect(userTeamIdsBson).headOption
```

Comment:
@fm3 Could we circumvent such errors by making the ids it's own type? Is this practical and can slick return a custom type?